File: sdk/python/evergreen_shims/exceptions.py
class EvergreenShimError(Exception):
    """Base exception for EvergreenShims SDK."""
    pass


class APIError(EvergreenShimError):
    """Raised when the API returns a non-success status code."""

    def __init__(self, status_code: int, message: str):
        self.status_code = status_code
        self.message = message
        super().__init__(f"API error {status_code}: {message}")


class ConnectionError(EvergreenShimError):
    """Raised when a connection to the server cannot be established."""
    pass


class TimeoutError(EvergreenShimError):
    """Raised when a request times out."""
    pass

File: sdk/node/src/exceptions.ts
export class EvergreenShimError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "EvergreenShimError";
  }
}

export class APIError extends EvergreenShimError {
  public statusCode: number;
  public body: string;

  constructor(statusCode: number, body: string) {
    super(`API error ${statusCode}: ${body}`);
    this.name = "APIError";
    this.statusCode = statusCode;
    this.body = body;
  }
}

export class ConnectionError extends EvergreenShimError {
  constructor(message: string) {
    super(message);
    this.name = "ConnectionError";
  }
}

export class TimeoutError extends EvergreenShimError {
  constructor(message: string) {
    super(message);
    this.name = "TimeoutError";
  }
}
